media-libs/openimageio/openimageio-1.7.17.recipe

```diff
@@ -0,0 +1,112 @@
+animation and visual effects work for film.
+OpenImageIO is used extensively in animation and VFX studios all over the \
+world, and is also incorporated into several commercial products."
+HOMEPAGE="http://github.com/OpenImageIO/"
+COPYRIGHT="2008-2017 Larry Gritz et al."
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="$HOMEPAGE/oiio/archive/Release-$portVersion.tar.gz"
+CHECKSUM_SHA256="669c59d06399dd882c3e2469bb3a4b33d537db150f6fe056495c3429b5c3838f"
+SOURCE_FILENAME="openimageio-$portVersion.tar.gz"
+SOURCE_DIR="oiio-Release-$portVersion"
+PATCHES="openimageio-$portVersion.patchset"
+
+ARCHITECTURES="x86 x86_64"
+SECONDARY_ARCHITECTURES="!x86_gcc2 x86"
+
+PROVIDES="
+	openimageio$secondaryArchSuffix = $portVersion
+	cmd:iconvert
+	cmd:idiff
+	cmd:igrep
+	cmd:iinfo
+	cmd:maketx
+	cmd:oiiotool
+	lib:libopenimageio$secondaryArchSuffix = $portVersion compat >= 1.7
+	lib:libopenimageio_util$secondaryArchSuffix = $portVersion compat >= 1.7
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libboost_atomic$secondaryArchSuffix
+	lib:libboost_chrono$secondaryArchSuffix
+	lib:libboost_date_time$secondaryArchSuffix
+	lib:libboost_filesystem$secondaryArchSuffix
+	lib:libboost_regex$secondaryArchSuffix
+	lib:libboost_system$secondaryArchSuffix
+	lib:libboost_thread$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libgif$secondaryArchSuffix
+	lib:libglew$secondaryArchSuffix
+	lib:libhalf$secondaryArchSuffix
+	lib:libilmimf$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libopencolorio$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix #yeah, really
+	lib:libtiff$secondaryArchSuffix
+	lib:libwebp$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	openimageio${secondaryArchSuffix}_devel = $portVersion
+	devel:libopenimageio$secondaryArchSuffix = $portVersion compat >= 1.7
+	devel:libopenimageio_util$secondaryArchSuffix = $portVersion compat >= 1.7
+	"
+REQUIRES_devel="
+	openimageio$secondaryArchSuffix == $portVersion base
+	devel:libboost_system$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libboost_system$secondaryArchSuffix
+	devel:libbz2$secondaryArchSuffix
+	devel:libcrypto$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libgif$secondaryArchSuffix
+	devel:libglew$secondaryArchSuffix
+	devel:libhalf$secondaryArchSuffix
+	devel:libilmimf$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libopencolorio$secondaryArchSuffix
+	devel:libpng16$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	devel:libtiff$secondaryArchSuffix
+	devel:libwebp$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	"
+
+BUILD()
+{
+	export CXXFLAGS="-std=c++11 -DBOOST_NO_CXX11_CONSTEXPR"
+	make USE_OPENSSL=1 $jobArgs
```

Comment:
could we use "cmake . -DUSE_OPENSSL=ON $cmakeDirArgs" plus "make install" instead?